### PR TITLE
16767 vet center branch remake

### PR DESCRIPTION
**Pull request by jtmst**
## Description
Relates to #[16767](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16767). (or closes?)
Remake of original PR as branch would not update for some reason

Adds in Components for vet center, hours and facility map
Adds queries, formaters,  and types for vet centers
Handles logic for handling centralized content in formatters

## Testing done
Local visual testing, playwrite tests added, unit tests added

## Screenshots
image

## QA steps
Verify vet-centers render as expected (ie: http://localhost:3000/pittsburgh-vet-center)
Verify storybook components render

```[tasklist]
- [ ] Automated tests have passed
- [ ] Tugboat environment was generated without errors
```

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:

### CODE REVIEW

**Review by tjheffner — DISMISSED** [review]
Overall, this is amazing work. Not a small content type by any means. Thank you for pushing on this!

I have some minor feedback, but I am happy to push them all as "fast follows" to allow 
this work to get merged.

The `On this page` and the facility locator map appear to be missing.
Screenshot 2024-03-13 at 7 43 41 AM

The `Need help after hours?` alert should be gray instead of blue
Screenshot 2024-03-13 at 7 43 55 AM

The feedback button at the bottom is missing (`ContentFooter` component in next-build)
Screenshot 2024-03-13 at 7 44 09 AM

Also noting that the accordion button text is showing undefined, but I think that's a bug that has been reported to the DST already

File: src/data/queries/index.ts [hunk 1]
```diff
@@ -51,17 +52,20 @@ export const QUERIES_MAP = {
   'node--story_listing': StoryListing,
   'node--q_a': QuestionAnswer,
   'node--event': Event,
+  'node--vet_center_facility_health_servi': HealthServices,
```
Comment (tjheffner) [inline src/data/queries/index.ts:55]: i'd move this down to near node--vet_center, for consistency

File: src/data/queries/index.ts [hunk 2]
```diff
@@ -51,17 +52,20 @@ export const QUERIES_MAP = {
...
   'node--support_service': SupportServices,
   'node--support_resources_detail_page': ResourcesSupport,
+  'node--vet_center': VetCenter,
+
   // Paragraphs
   'paragraph--alert': Alert,
   'paragraph--alert_single': AlertSingle,
-  'paragraph--audience_topics': AudienceTopics,
   'paragraph--basic_accordion': Accordion,
+  'paragraph--audience_topics': AudienceTopics,
```
Comment (tjheffner) [inline src/data/queries/index.ts:68]: keep this above basic_accordion, more alphabetical based on paragraph name, not component

File: src/templates/layouts/vetCenter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+    openingHoursSpecification: officeHours.map((hours) => ({
+      '@type': 'OpeningHoursSpecification',
+      dayOfWeek: `https://schema.org/${hours.day}`,
+      opens: hours.starthours,
+      closes: hours.endhours,
+    })),
+    hasMap: `https://maps.google.com?saddr=Current+Location&daddr=${encodeURIComponent(
+      `${address.address_line1}, ${address.locality}, ${address.postal_code}`
+    )}`,
+    // "image": [media.entity.image.derivative.url],
```
Comment (tjheffner) [inline src/templates/layouts/vetCenter/index.tsx:60]: why is this commented out? `image` is passed above, something like `image.links.2_1_large.href` likely has a valid url

File: src/templates/layouts/vetCenter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+      <>
+        <h2
+          id="prepare-for-your-visit"
+          className="vads-u-margin-top--0 vads-u-font-size--lg small-screen:vads-u-font-size--xl vads-u-margin-bottom--2"
+        >
+          Prepare for your visit
+        </h2>
+        <p>Click on a topic for more details.</p>
+        <div className="vads-u-margin-bottom--3">
+          {/* TODO: Look at this again after paragraph/basic_accordion work is complete */}
```
Comment (tjheffner) [inline src/templates/layouts/vetCenter/index.tsx:146]: is this todo still necessary after the other accordion PRs?

File: src/templates/layouts/vetCenter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+              <div>
+                <div className="vads-c-facility-detail">
+                  <section className="vads-facility-detail">
+                    <h3 className="vads-u-font-size--lg vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">
+                      Main Location
+                    </h3>
+
+                    {/* For the ExpandableOperatingStatus widget in vets-website */}
+
+                    {/* TODO: potential change to the vets-website widget to not use facilityId as a prop on div */}
```
Comment (tjheffner) [inline src/templates/layouts/vetCenter/index.tsx:204]: can you explain further here? it looks like you have the `facilityId` based on the ImageAndStaticMap included below

```
            <ImageAndStaticMap
              image={image}
              facilityId={fieldFacilityLocatorApiId}
            />
```

File: src/templates/layouts/vetCenter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+            >
+              <WysiwygField html={ccNonTraditionalHours.html} />
+            </div>
+          )}
+
+          {/* Call Center Information */}
+          {ccVetCenterCallCenter && (
+            <div className="vads-u-margin-bottom--2">
+              <AlertBlock
+                alertType="info"
```
Comment (tjheffner) [inline src/templates/layouts/vetCenter/index.tsx:273]: I called out the alert color, but it looks like the gray alert is removed in V3, so nevermind: https://design.va.gov/components/alert

File: src/types/drupal/node.ts [hunk 7]
```diff
@@ -158,6 +165,32 @@ export interface NodeHealthCareLocalFacility extends DrupalNode {
   field_operating_status_facility: string
   field_region_page: NodeHealthCareRegionPage
 }
+export interface NodeVetCenter extends DrupalNode {
+  field_address: FieldAddress
+  field_cc_non_traditional_hours: FieldCCText
+  field_cc_vet_center_call_center: FieldCCText
+  //TODO Fix after faq work
```
Comment (tjheffner) [inline src/types/drupal/node.ts:172]: are these TODOs still valid after the other accordion PRs? Do we know the types now?

File: src/templates/layouts/vetCenter/index.tsx (hunk 5 above)
Comment (jtmst) [reply src/templates/layouts/vetCenter/index.tsx:204]: In that case its being passed as a prop to a react component, but in this case its being used as the expected key on an html element.  my thought was that this should instead be a data attribute, but that would require a vets-website (and possibly content-buiild) change.

File: src/types/drupal/node.ts (hunk 7 above)
Comment (jtmst) [reply src/types/drupal/node.ts]: Yes. added those types and refactored the prepareForVisit section now.

**Review by tjheffner — APPROVED** [review]

**Pull request merged**